src/ua.c

```diff
@@ -1368,6 +1368,45 @@ void ua_hangup(struct ua *ua, struct call *call,
 }
 
 
+/**
+ * Redirect an incoming call
+ *
+ * @param ua     User-Agent
+ * @param call   Call to redirect, or NULL for current call
+ * @param scode  Status code
+ * @param reason Optional reason
+ * @param contact_params Body of Contact header
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+void ua_redirect(struct ua *ua, struct call *call,
+		 uint16_t scode, const char *reason,
+		 const char *contact_params)
```

Comment:
is this ua wrapper really needed ?

if the application has the `struct call` state it should have enough knowledge about it
to take the decision about redirect or not.

Reply:
That is good point and would greatly simplify my life, since I would need to hassle with the test case.  The only piece that currently prevents it is that I have not found an API call to test if call is answered.  In the PR there is new core.h function
```
bool call_answered(const struct call *call);
```
Would it be OK if I move it from core.h to baresip.h?

Reply:
there is no hard requirement to add new testcases.

Adding new testcase can be done if you have time and knowledge.
It can also be added later.

moving API functions from core.h to baresip.h is also fine